XunitRetry.Generator.SpecflowPlugin/Xunit2TestGeneratorProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Linq;
+using TechTalk.SpecFlow.Generator;
+using TechTalk.SpecFlow.Generator.UnitTestProvider;
+using TechTalk.SpecFlow.Utils;
+
+namespace XunitRetry.Generator.SpecflowPlugin
+{
+    public class XUnit2TestGeneratorProvider : XUnitTestGeneratorProvider
+    {
```

Comment:
@giggio I simply copied the Xunit[2]TestGeneratorProvider code directly from the SpecFlow repository (and made a few minor edits as required) - and I kept the same class structure as they had: https://github.com/techtalk/SpecFlow/blob/396e0e40ca06e68c01a4ca46fa0dfc9fbbb5a8c3/TechTalk.SpecFlow.Generator/UnitTestProvider/XUnit2TestGeneratorProvider.cs

I'm happy to refactor it in to one class. However, it means that if we want to take any future updates to these classes from the SpecFlow source, it will be extra work to factor the changes into our modified class hierarchy - whereas now it is a simple copy + paste. In any case I'm happy to refactor in to one class if that's what you'd prefer?